controllers/templatesync/template_sync.go

```diff
@@ -676,6 +678,17 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 			continue
 		}
 
+		// check for hub template error
+		if errAnno := metaObj.GetAnnotations()[hubTmplErrorKey]; errAnno != "" {
```

Comment:
If I'm reading this correctly, the ConfigurationPolicy that already exists would remain on the cluster even if there was a hub template error. This was also previously true in previous OCM versions, however, the config-policy-controller would not process these invalid ConfigurationPolicies. So I think this may be an unintended change in behavior.

Reply:
You're right. And your comment exposes the bug in the implementation just before this commit: while it prevents creation when there is a hub-template error, it doesn't remove an existing instance.

So it seems like it should delete the existing object in this situation. But now I'm wondering if we need to be very careful with configuration policies here, because of PruneObjectBehavior? What do you think

Reply:
@JustinKuli good point. It's hacky, but we could unset `pruneObjectBehavior` before deleting it. You lose the metadata on the status but at least it doesn't delete the object.

Alternatively, we could have template sync not handle ConfigurationPolicy hub template errors but handle all other policy kinds. Then the config-policy-controller would continue to handle them like before.

Reply:
Newly pushed change deals with it in the template-sync, and specially sets `pruneObjectBehavior` on ConfigurationPolicies in this situation.